crates/ruff_python_formatter/src/statement/stmt_assert.rs

```diff
@@ -7,6 +10,29 @@ pub struct FormatStmtAssert;
 
 impl FormatNodeRule<StmtAssert> for FormatStmtAssert {
     fn fmt_fields(&self, item: &StmtAssert, f: &mut PyFormatter) -> FormatResult<()> {
-        write!(f, [not_yet_implemented(item)])
+        let StmtAssert {
+            range: _,
+            test,
+            msg,
+        } = item;
+
+        write!(f, [text("assert"), space()])?;
+
+        if let Some(msg) = msg {
+            write!(
+                f,
+                [group(&format_args![
+                    test.format().with_options(Parenthesize::IfBreaks),
+                    text(","),
+                    space(),
+                    // `msg` gets parentheses if expanded so we don't need any beyond that.
+                    parenthesize_if_expands(&msg.format().with_options(Parenthesize::Never))
```

Comment:
Is this different from `msg.format().with_options(Parenthesize::IfBreaks)`?

Reply:
Yea I'm definitely not satisfied with it yet.

Without my "hack" here I would get 

```
black_compatibility@simple_cases__composition_no_trailing_comma.py.snap
  349       │-        } == expected, (
  350       │-            "Not what we expected and the message is too long to fit in one line"
  351       │-        )
        360 │+        } == expected, "Not what we expected and the message is too long to fit in one line"
```
`black` will wrap the `msg` (lines 349-351). So IIUC `Parenthesize::IfBreaks` wouldn't give me that alone since the `msg` doesn't break as a long string like that.

> It may also be that e.g. this is caused by incorrect formatting of string

I don't know enough yet but maybe this is possible.

Reply:
> My understanding from [Black's code](https://github.com/psf/black/blob/d1248ca9beaf0ba526d265f4108836d89cf551b7/src/black/linegen.py#L484-L499) is that assert should work the same as for if headers

This could be helpful. I'll check it out. Thank you!

Reply:
Let's see if https://github.com/astral-sh/ruff/pull/5708 helps

Reply:
I spent some more time reading black's source code and what I think we miss is that strings have a higher split priority than any math operator. I'm not sure how we want to model this yet but I suggest to not block this PR because of it.